lib/logstash/outputs/opensearch/http_client.rb

```diff
@@ -379,11 +383,19 @@ def exists?(path, use_get=false)
     end
 
     def template_exists?(name)
-      exists?("/#{template_endpoint}/#{name}")
+      if legacy_template
```

Comment:
It might be better to keep this unchanged and move the logic into the template_endpoint method